[julius_ros] Update julius to 4.4.2 / add ROS interface

furushchev opened the pull request.

k-okada commented:
can we add msg/srv to https://github.com/jsk-ros-pkg/jsk_common_msgs/tree/master/speech_recognition_msgs?

furushchev commented:
@k-okada That's good idea. My worry is that `Vocabulary.msg` is julius specific. (So `SpeechRecognition.srv` which uses `Vocabulary.msg` internally can be said also julius specific)
Can we still add to `speech_recognition_msgs`?

k-okada commented:
What is the definition of Vicabulary?

2017年5月17日(水) 13:26 Furushchev <notifications@github.com>:

> @k-okada <https://github.com/k-okada> That's good idea. My worry is that
> Vocabulary.msg is julius specific. (So SpeechRecognition.srv which uses
> Vocabulary.msg internally can be said also julius specific)
> Can we still add to speech_recognition_msgs?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/99#issuecomment-301981871>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3A-Yt1_8PstF0fUNonYnRPRgMNtQks5r6nbhgaJpZM4NckjZ>
> .
>
-- 
--
◉ Kei Okada

furushchev commented:
@k-okada Vocabulary contains two slots: `words` and `phonemes`.

Phomenes are phonemes for each words, for example, 

words: かけ, きつね
phonemes: k a k e, k i ts u n e

If `words` are filled but `phonemes` are left empty, `julius_client.py` node automatically estimates phonemes from words (but only if words are written in ひらがな)

k-okada commented:
I think that's ok, so let's add jsk_common_msgs

2017年5月17日(水) 15:17 Furushchev <notifications@github.com>:

> @k-okada <https://github.com/k-okada> Vocabulary contains two slots: words
> and phonemes.
>
> Phomenes are phonemes for each words, for example,
>
> words: かけ, きつね
> phonemes: k a k e, k i ts u n e
>
> If words are filled but phonemes are left empty, julius_client.py node
> automatically estimates phonemes from words (but only if words are written
> in ひらがな)
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/99#issuecomment-301996408>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3Fnrx3E7bNPgf2sFWJ2hotsfwYSdks5r6pEKgaJpZM4NckjZ>
> .
>
-- 
--
◉ Kei Okada

furushchev commented:
@k-okada Thanks for review!

- Moved msg / srv files to `speech_recognition_msgs`.
https://github.com/jsk-ros-pkg/jsk_common_msgs/pull/15

Updated commits

k-okada commented:
released https://github.com/ros/rosdistro/pull/14997

--
◉ Kei Okada


2017-05-17 18:56 GMT+09:00 Furushchev <notifications@github.com>:

> @k-okada <https://github.com/k-okada> Thanks for review!
>
>    - Moved msg / srv files to speech_recognition_msgs.
>    jsk-ros-pkg/jsk_common_msgs#15
>    <https://github.com/jsk-ros-pkg/jsk_common_msgs/pull/15>
>
> Updated commits
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/99#issuecomment-302043760>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3Oduu-EXUWiBL4_mecOaBkKX0ahXks5r6sRYgaJpZM4NckjZ>
> .
>

snozawa commented:
Nice work!!
I have two questions:
- In the case of using julius package from deb, is julius somthing (hmm models ...etc) installed via `ros-indigo-julius`?
- In the case of using julius package from source, is `<build_depend>julius</build_depend>`  required in package.xml in `julius_ros`?

furushchev commented:
@k-okada Thanks!

furushchev commented:
@snozawa 

> In the case of using julius package from deb, is julius somthing (hmm models ...etc) installed via ros-indigo-julius?

I updated both `julius` and `julius_ros` packages at this pull request, so if you want to use `julius_ros` right now, you just will have to checkout this branch and run `catkin build julius_ros` which also builds `julius` as dependency.
Configuration files (you mentioned as hmm models...etc) are also installed into devel / install space when `julius` is built.
If not so, please let me know, there might be bug. :+1: 

> In the case of using julius package from source, is <build_depend>julius</build_depend> required in package.xml in julius_ros?

Currently any source codes at `julius_ros` package does NOT need `julius` package on build time, so I think we don't have to add `julius` as `build_depend`.(It is still `run_depend` so if both `julius` and `julius_ros` are in catkin work space, they will be both built.)

snozawa commented:
Thanks!!
I understand information about `ros-indigo-julius`.

> Currently any source codes at julius_ros package does NOT need julius package on build time, so I think we don't have to add julius as build_depend.(It is still run_depend so if both julius and julius_ros are in catkin work space, they will be both built.)

I see.
In my one PC, we need to write `build_depend` on `julius`.
However, in another PC, we do not need to write `build_depnd` on `julius` and I successfully built `julius_ros` without any modification.
This seems to be difference of versions of catkin related packages.

Setting of the PC I succeeded:
```
$ pip freeze | grep catkin
catkin==0.6.19
catkin-lint==1.4.8
catkin-pkg==0.2.10
catkin-tools==0.4.4
```
Setting of the PC I failed:
```
pip freeze | grep catkin
catkin==0.6.19
catkin-pkg==0.2.10
catkin-tools==0.3.1
You are using pip version 8.1.1, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

snozawa commented:
(Related with off-line talk with @furushchev)
I have two suggestions:

- To launch julius softwares, there are two ways: include `julius.launch` or write user-own-code by copying `julius.launch`. Because the current `julius.launch` is well-written, including `julius.launch` is convenient for me. If so, please add argument to specify `sound_play` launching (true/false), because some robot users have already launched `sound_play`.
- Current `sound_play` node assumes `sound_play` as topic name. On the other hand, current jsk users (e.g., `speak.l` users) assumes `robotsound` as topic name. It would be nice to add instruction to add `remap` or something to REAME.md.

furushchev commented:
Test failed. Maybe bug of `catkin_tools`??

```bash
Traceback (most recent call last):
  File "/usr/bin/catkin", line 9, in <module>
    load_entry_point('catkin-tools==0.4.4', 'console_scripts', 'catkin')()
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 267, in main
    catkin_main(sysargs)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/commands/catkin.py", line 262, in catkin_main
    sys.exit(args.main(args) or 0)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/cli.py", line 420, in main
    summarize_build=opts.summarize  # Can be True, False, or None
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 286, in build_isolated_workspace
    built_packages, unbuilt_pkgs = get_built_unbuilt_packages(context, workspace_packages)
  File "/usr/lib/python2.7/dist-packages/catkin_tools/verbs/catkin_build/build.py", line 151, in get_built_unbuilt_packages
    find_packages(context.package_metadata_path(), warnings=[]).items()])
  File "/usr/lib/python2.7/dist-packages/catkin_pkg/packages.py", line 87, in find_packages
    packages = find_packages_allowing_duplicates(basepath, exclude_paths=exclude_paths, exclude_subspaces=exclude_subspaces, warnings=warnings)
  File "/usr/lib/python2.7/dist-packages/catkin_pkg/packages.py", line 131, in find_packages_allowing_duplicates
    path_parsed_packages, warnings_lists = zip(*multiprocessing.Pool().map(parser, data))
ValueError: need more than 0 values to unpack
/home/travis/.travis/job_stages: line 155: 24069 Terminated
```

@k-okada Could you restart to confirm? (since I could not restart)

furushchev commented:
Added test / bagdata of speech and passed the test!
@k-okada Please merge if there is no problem.
